src/app/Shared/Services/CommandChannel.service.tsx

```diff
@@ -27,21 +27,27 @@ export class CommandChannel {
         (url: any) => this.clientUrlSubject.next(url.clientUrl),
         (err: any) => this.logError('Client URL configuration', err)
       );
+    
+    const getDatasourceURL = fromFetch(`${this.apiSvc.authority}/api/v1/grafana_datasource_url`)
+    .pipe(concatMap(resp => from(resp.json())));
+    const getDashboardURL = fromFetch(`${this.apiSvc.authority}/api/v1/grafana_dashboard_url`)
+    .pipe(concatMap(resp => from(resp.json())));
 
-    fromFetch(`${this.apiSvc.authority}/api/v1/grafana_datasource_url`)
-      .pipe(concatMap(resp => from(resp.json())))
-      .subscribe(
-        (url: any) => this.grafanaDatasourceUrlSubject.next(url.grafanaDatasourceUrl),
-        (err: any) => this.logError('Grafana Datasource configuration', err)
-      );
-
-    fromFetch(`${this.apiSvc.authority}/api/v1/grafana_dashboard_url`)
-      .pipe(concatMap(resp => from(resp.json())))
+    fromFetch(`${this.apiSvc.authority}/health`)
+      .pipe(
+        concatMap(resp => from(resp.json())))
+      .pipe(
+        concatMap(jsonResp => {
+            if((jsonResp.dashboardAvailable == true) && (jsonResp.datasourceAvailable == true)) 
+              {return forkJoin([getDatasourceURL, getDashboardURL])}
+            else {this.logError('Grafana configuration', 'URLs unavailable');
+                  return jsonResp}}))
```

Comment:
I'm a little confused on this - in the `if` branch we're returning an Observable with the `forkJoin` operator. But in the `else` branch, this looks like it's returning a simple JSON object from the HTTP response, isn't it?

I think the error handling can be cleaned up here by returning the same `forkJoin` in the `if` branch, and returning a `throw` operator in this `else` branch (and not logging the error, yet). Then, the subscriber below should gain an error handler, and this is what should call `logError`.